src/main/java/com/bean00/Server.java

```diff
@@ -5,21 +5,22 @@
 import java.io.Writer;
 
 public class Server {
+    private ResponseWriter responseWriter = new ResponseWriter();
 
-    public void run(MessageController messageController,
+    public void run(RequestProcessor requestProcessor,
                     BufferedReader in, Writer out) throws IOException {
```

Comment:
I like the direction this is heading, but I notice now that `in` and `out` are only passed in so they can be passed to the `ResponseWriter` and the `RequestParser`. Is there benefit to `Server` having direct access to `in` and `out`? Or are they just passing through?

Reply:
Right now they are just passing through. It would be nice to be able to pass `in` and `out` directly to the objects / methods that need them, but I can't really think of a way to do that (`in` and `out` will still need to be passed in to `Server` somehow, if I want to keep the objects that use them at a lower level). I am open to suggestions though, on how to improve this.

Reply:
What about passing the `RequestParser` and the `ResponseWriter` to `Server` instead of `in` and `out`?

Reply:
Done. It actually worked out pretty well, and made the code cleaner and more expressive.